arrows/ffmpeg/ffmpeg_video_input.cxx

```diff
@@ -0,0 +1,826 @@
+      {
+        this->f_data_index = i;
+      }
+    }
+
+    if (this->f_video_index < 0)
+    {
+      LOG_ERROR(this->logger, "Error: could not find a video stream in " << this->video_path);
+      return false;
+    }
+
+    if (this->f_data_index < 0)
+    {
+      LOG_INFO(this->logger, "No data stream available, using AVMEDIA_TYPE_UNKNOWN stream instead");
+      // Fallback for the DATA stream if incorrectly coded as UNKNOWN.
+      for (unsigned i = 0; i < this->f_format_context->nb_streams; ++i)
+      {
+        AVCodecContext *enc = this->f_format_context->streams[i]->codec;
+        if (enc->codec_type == AVMEDIA_TYPE_UNKNOWN)
+        {
+          this->f_data_index = i;
+        }
+      }
+    }
+
+    av_dump_format(this->f_format_context, 0, this->video_path.c_str(), 0);
+
+    // Open the stream
+    AVCodec* codec = avcodec_find_decoder(codec_context_origin->codec_id);
+    if (!codec)
+    {
+      LOG_ERROR(this->logger,
+        "Error: Codec " << codec_context_origin->codec_descriptor
+        << " (" << codec_context_origin->codec_id << ") not found");
+      return false;
+    }
+
+    // Copy context
+    this->f_video_encoding = avcodec_alloc_context3(codec);
+    if (avcodec_copy_context(this->f_video_encoding, codec_context_origin) != 0)
+    {
+      LOG_ERROR(this->logger, "Error: Could not copy codec " << this->f_video_encoding->codec_id);
+      return false;
+    }
+
+    // Open codec
+    if (avcodec_open2(this->f_video_encoding, codec, NULL) < 0)
+    {
+      LOG_ERROR(this->logger, "Error: Could not open codec " << this->f_video_encoding->codec_id);
+      return false;
+    }
+
+    this->f_video_stream = this->f_format_context->streams[this->f_video_index];
+    this->f_frame = av_frame_alloc();
+
+    if (this->f_video_stream->start_time == int64_t(1) << 63)
+    {
+      this->f_start_time = 0;
+    }
+    else
+    {
+      this->f_start_time = this->f_video_stream->start_time;
+    }
+
+    // The MPEG 2 codec has a latency of 1 frame when encoded in an AVI
+    // stream, so the pts of the last packet (stored in pts) is
+    // actually the next frame's pts.
+    if (this->f_video_stream->codec->codec_id == AV_CODEC_ID_MPEG2VIDEO &&
+      std::string("avi") == this->f_format_context->iformat->name)
+    {
+      this->f_frame_number_offset = 1;
+    }
+
+    // Not sure if this does anything, but no harm either
+    av_init_packet(&this->f_packet);
+    this->f_packet.data = nullptr;
+    this->f_packet.size = 0;
+
+    return true;
+  }
+
+  // ==================================================================
+  /*
+  * @brief Advance to the next frame (but don't acquire an image).
+  *
+  * @return \b true if video was valid and we found a frame.
+  */
+  virtual bool advance()
+  {
+    // Quick return if the file isn't open.
+    if (!this->is_opened())
+    {
+      this->frame_advanced = 0;
+      return false;
+    }
+
+    // \todo - num_frames not implemented
```

Comment:
I think this should just be removed since `num_frames` is not implemented. It probably makes sense to add a TODO or better yet open an issue to remind us to improved how the number of frames is reported but this comment is confusing since `num_frames` is not available.